tools/protoxform/protoprint.py

```diff
@@ -580,6 +622,27 @@ def visit_service(self, service_proto, type_context):
     def visit_enum(self, enum_proto, type_context):
         if protoxform_options.has_hide_option(enum_proto.options):
             return ''
+        # Verify that not hidden deprecated enum values of non-frozen protos have valid version
+        # annotations.
+        for v in enum_proto.value:
+            if v.options.deprecated and not self._frozen_proto and \
+                    not protoxform_options.has_hide_option(v.options):
+                # If the enum value has annotation from deprecation.proto, need to import it.
+                if v.options.HasExtension(deprecation_pb2.deprecated_at_minor_version_enum) or \
+                        v.options.HasExtension(deprecation_pb2.disallowed_by_default_enum):
+                    self._needs_deprecation_annotation_import = True
+                if v.name != 'DEPRECATED_AND_UNAVAILABLE_DO_NOT_USE':
+                    if v.options.HasExtension(deprecation_pb2.deprecated_at_minor_version_enum):
+                        if not validate_deprecation_version(v.options.Extensions[
+                                deprecation_pb2.deprecated_at_minor_version_enum]):
+                            raise ProtoPrintError(
+                                'Error while parsing "deprecated_at_minor_version_enum" annotation value for enum value %s.'
+                                % v.name)
+                    else:
+                        self._needs_deprecation_annotation_import = True
+                        v.options.Extensions[
+                            deprecation_pb2.
+                            deprecated_at_minor_version_enum] = self._deprecated_annotation_version_value
```

Comment:
Suggest splitting into an independent method, the logic is complex enough.

Reply:
Created a new function. Done.